lib/headers.ml

```diff
@@ -43,24 +43,20 @@ let of_list t = of_rev_list (List.rev t)
 let to_rev_list t = t
 let to_list t = List.rev (to_rev_list t)
 
-exception Local
-
 module CI = struct
   let lower c =
     if c >= 0x41 && c <= 0x5a then c + 32 else c
 
   let equal x y =
     let len = String.length x in
-    len = String.length y &&
-    match for i = 0 to len - 1 do
-      let c1 = Char.code (String.unsafe_get x i) in
-      let c2 = Char.code (String.unsafe_get y i) in
-      if c1 = c2
-      then ()
-      else if lower c1 <> lower c2 then raise Local
-    done with
-    | () -> true
-    | exception Local -> false
+    let rec equal_aux i =
```

Comment:
This is going to allocate a closure due to `len`, `x`, and `y` being free in the body of the loop, which will negatively affect performance on x86. It should be possible to a avoid this by passing them as arguments. You could even eliminate `len` entirely by walking the strings backwards.

Reply:
Makes sense, I'll try to fix this.

Reply:
I've updated the branch taking conservative approach: uplifting the closure to standalone function, which should ensure no allocations happen. Please take a look.